pkg/compact/downsample/downsample_test.go

```diff
@@ -563,10 +568,12 @@ func TestDownsampleAggrAndNonEmptyXORChunks(t *testing.T) {
 	mb.addSeries(ser)
 
 	fakeMeta := &metadata.Meta{}
+	// target
 	fakeMeta.Thanos.Downsample.Resolution = 300_000
+	// already existing resolution
 	id, err := Downsample(logger, fakeMeta, mb, dir, 3_600_000)
 	_ = id
-	testutil.NotOk(t, err)
+	testutil.Ok(t, err)
```

Comment:
Can we test that we downsample the XOR chunks to aggregated chunks?

Reply:
@yeya24 sure, I'll add that in test cases.
I also wanted to know, in the solution suggested by you, you had mentioned making changes in the `bucket tools repair` but I feel it can be handled in the way we have here. 
Also you have mentioned that we have to do that only if XOR chunks contain `enough samples` how many samples would be enough samples? Is there any predefined threshold that defines what enough samples are? 
Thanks a lot for reviewing

Reply:
Handling this in the downsample component itself is also good to me.

> Also you have mentioned that we have to do that only if XOR chunks contain enough samples how many samples would be enough samples?

This should be fine. We downsample block when the block has a large block duration. So if the block is being downsampled to 1h, which means the block is already large. And even if the chunks are small, `downsampleRaw` should handle the situation correctly.

Reply:
@yeya24 Seems like everything is fine, I have made changes in the test cases as well, please review them again, and if everything seems fine, I will add it into the `changelog` and remove the `WIP` tag.
Thanks a lot for the already made reviews, glad you liked the solution :smile: